dotrix_core/src/renderer/mapped_wgpu.rs

```diff
@@ -0,0 +1,416 @@
+        Self {
+            read: true,
+            write: true,
+            as_texture: true,
+            as_storage: true,
+            as_render: true,
+        }
+    }
+    /// Create a texture with no usage
+    #[must_use]
+    pub fn none() -> Self {
+        Self {
+            read: false,
+            write: false,
+            as_texture: false,
+            as_storage: false,
+            as_render: false,
+        }
+    }
+    /// Start creating a texture usage desciption
+    #[must_use]
+    pub fn create() -> Self {
+        Self::none()
+    }
+    /// Allow the texture's gpu buffer to be read from
+    #[must_use]
+    pub fn read(mut self) -> Self {
+        self.read = true;
+        self
+    }
+    /// Allow the texture's gpu buffer to be written to
+    #[must_use]
+    pub fn write(mut self) -> Self {
+        self.write = true;
+        self
+    }
+    /// Allow the texture's gpu buffer to be used as a render attachment
+    #[must_use]
+    pub fn render(mut self) -> Self {
+        self.as_render = true;
+        self
+    }
+    /// Allow the texture's gpu buffer to be bound as a StorageTexture
+    #[must_use]
+    pub fn storage(mut self) -> Self {
+        self.as_storage = true;
+        self
+    }
+    /// Allow the texture's gpu buffer to be bound as a Texture
+    #[must_use]
+    pub fn texture(mut self) -> Self {
+        self.as_texture = true;
+        self
+    }
+}
+
+impl From<TextureUsages> for wgpu::TextureUsages {
+    fn from(obj: TextureUsages) -> wgpu::TextureUsages {
+        let mut res = wgpu::TextureUsages::empty();
+        if obj.read {
+            res |= wgpu::TextureUsages::COPY_SRC
+        }
+        if obj.write {
+            res |= wgpu::TextureUsages::COPY_DST
+        }
+        if obj.as_render {
+            res |= wgpu::TextureUsages::RENDER_ATTACHMENT
+        }
+        if obj.as_storage {
+            res |= wgpu::TextureUsages::STORAGE_BINDING
+        }
+        if obj.as_texture {
+            res |= wgpu::TextureUsages::TEXTURE_BINDING
+        }
+        res
+    }
+}
+
+/// Defines the possible access modes for a
+/// storage texture.
+#[derive(Copy, Clone, Debug)]
+pub enum StorageTextureAccess {
+    /// Read access
+    Read,
+    /// Write access
+    Write,
+    /// Read write access
+    ReadWrite,
+}
+
+impl From<StorageTextureAccess> for wgpu::StorageTextureAccess {
+    fn from(obj: StorageTextureAccess) -> Self {
+        match obj {
+            StorageTextureAccess::Read => Self::ReadOnly,
+            StorageTextureAccess::Write => Self::WriteOnly,
+            StorageTextureAccess::ReadWrite => Self::ReadWrite,
+        }
+    }
+}
+
+/// The texture format used while on the gpu
+#[derive(Copy, Clone, Debug)]
+pub struct TextureFormat {
+    /// The raw `[wgpu::TextureFormat]`
+    pub wgpu_texture_format: WgpuTextureFormat,
+    /// Defines if the channel data is filterable with a filtering sampler
+    pub filterable: bool,
```

Comment:
Lets make a getter instead of property?
```
pub fn is_filterable(&self) -> bool {
   self. wgpu_texture_format.describe().guaranteed_format_features.filterable
}
```
?

Reply:
Ok Iv'e made this change, I decided to make it pub(create) because there should not be much need for the end user to see it and we might want to change the interface in case we decided to implement the `Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES` feature which would allow us to filter f32 even though `self. wgpu_texture_format.describe().guaranteed_format_features.filterable` would say false.